Rx/v2/src/rxcpp/schedulers/rx-runloop.hpp

```diff
@@ -79,6 +80,9 @@ struct run_loop_scheduler : public scheduler_interface
                 std::unique_lock<std::mutex> guard(st->lock);
                 st->q.push(detail::run_loop_state::item_type(when, scbl));
                 st->r.reset(false);
+                std::function<void()> callback{st->add_task_callback};
+                guard.unlock(); // So we can't get attempt to recursively lock the state
+                if (callback) callback();
```

Comment:
`schedule()` is called from threads other than the thread the owns the `runloop`.

`callback()` is also called from other threads. The callback must not dispatch `runloop` items. The only thing the callback could safely do is set a timer to trigger at an earlier time than it currently is set.

passing the new `when` would make that easier.
only calling the callback if the new time is earlier than the previous top item would be reasonable as well.

If those changes are made then the name of the callback might change as well. Perhaps `notify_when_changed`?